### PR TITLE
gh-126014: Ignore `__pycache__`-only folders in makefile tests

### PR DESCRIPTION
**Pull request by sobolevn**
* Issue: gh-126014

### CODE REVIEW

**Review by furkanonder — APPROVED** [review]
LGTM.

File: Lib/test/test_tools/test_makefile.py [hunk 1]
```diff
@@ -52,7 +52,10 @@ def test_makefile_test_folders(self):
             if not dirs and not files:
                 continue
             # Skip dirs with hidden-only files:
-            if files and all(filename.startswith('.') for filename in files):
+            if files and all(
+                filename.startswith('.') or filename in {'__pycache__'}
```
Comment (tomasr8) [inline Lib/test/test_tools/test_makefile.py]: Maybe a direct comparison is better here? (unless we're planning to ignore more filenames at some point)
```suggestion
                filename.startswith('.') or filename == '__pycache__'
```

**Review by tomasr8 — APPROVED** [review]
Thank you! :) I ran into this again today while working on some tests :smile:

File: Lib/test/test_tools/test_makefile.py (hunk 1 above)
Comment (sobolevn) [reply Lib/test/test_tools/test_makefile.py]: I don't think we are really planning any extras :)
So, yeah!

**Pull request merged**